src/pair_mie_cut.f90

```diff
@@ -0,0 +1,131 @@
+    procedure :: setup => pair_mie_cut_setup
+    procedure :: compute => pair_mie_cut_compute
+    procedure :: virial => pair_mie_cut_virial
+    procedure :: mix => pair_mie_cut_mix
+end type pair_mie_cut
+
+contains
+
+!---------------------------------------------------------------------------------------------------
+
+  subroutine pair_mie_cut_setup( model, params, iparams )
+    class(pair_mie_cut), intent(inout) :: model
+    real(rb), intent(in), optional :: params(:)
+    integer,  intent(in), optional :: iparams(:)
+    real(rb) :: coeff
+    ! Model name:
+    model%name = "mie_cut"
+
+    ! Model parameters:
+    model%epsilon = params(1)
+    model%sigma = params(2)
+    model%n = params(3)
+    model%m = params (4)
+    ! Pre-computed quantities:
+    coeff  = (model%n / (model%n - model%m)) * (model%n / model%m) ** (model%m / (model%n - model%m)) * model%epsilon
+    model%mien = coeff * model%sigma ** model%n
+    model%miem = coeff * model%sigma ** model%m
+
+  end subroutine pair_mie_cut_setup
+
+!---------------------------------------------------------------------------------------------------
+
+  subroutine pair_mie_cut_compute( model, Eij, Wij, invR, invR2 )
+    class(pair_mie_cut), intent(in)  :: model
+    real(rb),           intent(out) :: Eij, Wij, invR
+    real(rb),           intent(in)  :: invR2
+
+    real(rb) :: rn, rm
+
+    rm = invr2 ** (model%m / 2.0_rb)
+    rn = rm ** (model%n / model%m)
+
+    Eij = model%mien * rn - model%miem * rm 
+    Wij = model%n * model%mien * rn - model%m * model%miem * rm
+
+  end subroutine pair_mie_cut_compute
+
+!---------------------------------------------------------------------------------------------------
+
+  subroutine pair_mie_cut_virial( model, Wij, invR, invR2 )
+    class(pair_mie_cut), intent(in)  :: model
+    real(rb),           intent(out) :: Wij, invR
+    real(rb),           intent(in)  :: invR2
+
+    real(rb) :: rn, rm
+
+    rm = invr2 ** (model%m / 2.0_rb)
+    rn = rm ** (model%n / model%m)
+
+    Wij = model%n * model%mien * rn - model%m * model%miem * rm
+
+  end subroutine pair_mie_cut_virial
+
+!---------------------------------------------------------------------------------------------------
+
+  function pair_mie_cut_mix( this, other ) result( mixed )
+    class(pair_mie_cut), intent(in) :: this
+    class(cPairModel),  intent(in) :: other
+    class(cPairModel),  pointer    :: mixed
+
+    select type (other)
+
+      class is (pair_mie_cut)
+        allocate(pair_mie_cut :: mixed)
+        call mixed % setup( [sqrt(this%epsilon*other%epsilon)*  &
+                            sqrt(this%sigma**3.0_rb*other%sigma**3.0_rb) & 
+                            /(half*(this%sigma + other%sigma)), &
+                             half*(this%sigma + other%sigma),  &
+                             3.0_rb + sqrt((this%n - 3.0_rb)*(other%n - 3.0_rb)),  &
+                             3.0_rb + sqrt((this%m - 3.0_rb)*(other%m - 3.0_rb))] )
+      class default
```

Comment:
Since Lennard-Jones is a special case of Mie, it would be nice to have a Mie-LJ mixing rule in which other%m and other%n are replaced by 12 and 6, respectively.